pkg/sql/sequence.go

```diff
@@ -208,3 +209,109 @@ func assignSequenceOptions(
 
 	return nil
 }
+
+// maybeAddSequenceDependency:
+// - if the given expr uses a sequence
+//   - adds a reference on `col` to the sequence. (mutates `col`; caller must persist changes to KV)
+//   - adds a reference on the sequence descriptor to `col`. (returns the modified sequence
+//     descriptor for the caller to save)
+func maybeAddSequenceDependency(
+	tableDesc *sqlbase.TableDescriptor,
+	col *sqlbase.ColumnDescriptor,
+	expr tree.TypedExpr,
+	evalCtx *tree.EvalContext,
+) (*sqlbase.TableDescriptor, error) {
+	ctx := evalCtx.Ctx()
+	seqName, err := getUsedSequenceName(expr)
+	if err != nil {
+		return nil, err
+	}
+	if seqName == nil {
+		return nil, nil
+	}
+	parsedSeqName, err := evalCtx.Planner.ParseQualifiedTableName(ctx, *seqName)
+	if err != nil {
+		return nil, err
+	}
+	seqDesc, err := getSequenceDesc(ctx, evalCtx.Txn, NilVirtualTabler, parsedSeqName)
+	if err != nil {
+		return nil, err
+	}
+	col.UsesSequenceId = &seqDesc.ID
+	// Now that we've gotten the sequence descriptor, I guess we can mutate it?
+	seqDesc.DependedOnBy = append(seqDesc.DependedOnBy, sqlbase.TableDescriptor_Reference{
+		ID:        tableDesc.ID,
+		ColumnIDs: []sqlbase.ColumnID{col.ID},
```

Comment:
what if there are multiple columns using the same sequence?

Reply:
`maybeAddSequenceDependency` gets called for each column, so it'll end up as a separate entry in `seqDesc.DependedOnBy` for each. Will add a logic test for this.